docs/docs-beta/scripts/generate-code-example-imports.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs');
+const path = require('path');
+
+const DOCUMENTATION_DIRECTORY = 'docs';
+const VALID_DOCUMENT_EXTENSIONS = ['.md', '.mdx'];
+const CODE_EXAMPLE_PATH_REGEX = /<CodeExample\s+[^>]*path=["']([^"']+)["'][^>]*>/g;
+
+/**
+ * Returns a list of file paths for a given `dir`.
+ */
+function getAllDocuments(dir) {
+  let results = [];
+  const list = fs.readdirSync(dir);
+
+  list.forEach((file) => {
+    file = path.join(dir, file);
+    const stat = fs.statSync(file);
+    if (stat && stat.isDirectory()) {
+      results = results.concat(getAllDocuments(file)); // Recurse into subdirectory
+    } else {
+      if (VALID_DOCUMENT_EXTENSIONS.indexOf(path.extname(file)) !== -1) {
+        results.push(file); // Add file to results
+      }
+    }
+  });
+
+  return results;
+}
+
+/**
+ * Extracts all `regex` group `1` matches found in a list of `files`.
+ */
+function getUniqueRegexMatches(files, regex) {
+  const matches = new Set();
+
+  files.forEach((file) => {
+    const content = fs.readFileSync(file, 'utf-8');
+    let foundMatches;
+    while ((foundMatches = regex.exec(content)) !== null) {
+      matches.add(foundMatches[1]); // Extract group 1
+    }
+  });
+
+  return Array.from(matches);
+}
+
+const files = getAllDocuments('docs');
+
+const uniqueMatches = getUniqueRegexMatches(files, CODE_EXAMPLE_PATH_REGEX);
+
+const lines = [
+  'export const CODE_EXAMPLE_PATH_MAPPINGS = {',
+  //...uniqueMatches.map((path) => `  '${path}': '!!raw-loader!/../../examples/${path}',`),
+  //...uniqueMatches.map((path) => `  '${path}': require('!!raw-loader!/../../examples/${path}'),`),
+  ...uniqueMatches.map((path) => `  '${path}': import('!!raw-loader!/../../examples/${path}'),`),
+  '};',
```

Comment:
Hmm, this import is going to happen immediately, which means every page will effectively load every single possible code example. To prevent that you should wrap this in a function and require explicitly calling it to make the import. I suspect this immediate import is why you don't need the server logic we needed before.

Reply:
Gotcha, I can look into wrapping it in a function, but will be logging off in a sec.

I'm curious if the import will still be a problem when the site is statically rendered with `yarn build`? Maybe we can chat sometime in the morning if that works for you!